File: backend/app/utils/riot_id.py
def parse_riot_id(riot_id: str) -> tuple[str, str]:
    """Split a Riot ID into game name and tag line (split on last #)."""
    trimmed = riot_id.strip()
    if "#" not in trimmed:
        raise ValueError("Riot ID must include a tag, e.g. Player#EUW")
    idx = trimmed.rindex("#")
    game_name = trimmed[:idx].strip()
    tag_line = trimmed[idx + 1 :].strip().lstrip("#")
    if not game_name or not tag_line:
        raise ValueError("Riot ID must include both name and tag")
    return game_name, tag_line

File: frontend/src/landing/app/lib/riotId.ts
export function parseRiotId(riotId: string): { gameName: string; tagLine: string } {
  const trimmed = riotId.trim();
  const idx = trimmed.lastIndexOf("#");
  if (idx <= 0 || idx === trimmed.length - 1) {
    throw new Error("Riot ID must include a tag, e.g. Player#EUW");
  }
  const gameName = trimmed.slice(0, idx).trim();
  const tagLine = trimmed.slice(idx + 1).trim();
  if (!gameName || !tagLine) {
    throw new Error("Riot ID must include both name and tag");
  }
  return { gameName, tagLine };
}
